demos/rag_agentic/notebooks/Level5_agentic_MCP_and_RAG.ipynb

```diff
@@ -0,0 +1,669 @@
+    "\n",
+    "if \"mcp::openshift\" not in registered_toolgroups: # required\n",
+    "    client.toolgroups.register(\n",
+    "        toolgroup_id=\"mcp::openshift\",\n",
+    "        provider_id=\"model-context-protocol\",\n",
+    "        mcp_endpoint={\"uri\":ocp_mcp_url},\n",
+    "    )\n",
+    "\n",
+    "if \"mcp::slack\" not in registered_toolgroups: # required\n",
+    "    client.toolgroups.register(\n",
+    "        toolgroup_id=\"mcp::slack\",\n",
+    "        provider_id=\"model-context-protocol\",\n",
+    "        mcp_endpoint={\"uri\":slack_mcp_url},\n",
+    "    )\n",
+    "\n",
+    "# Log the current toolgroups registered\n",
+    "logger.info(\n",
+    "    f\"Your Llama Stack server is already registered with the following tool groups: {set(registered_toolgroups)}\\n\"\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2bf70ace-b704-4379-aa88-3c793cc4f959",
+   "metadata": {},
+   "source": [
+    "### System Prompts for different models\n",
+    "\n",
+    "**Note:** If you have multiple models configured with your Llama Stack server, you can choose which one to run your queries against. When switching to a different model, you may need to adjust the system prompt to align with that model’s expected behavior. Many models provide recommended system prompts for optimal and reliable outputs these are typically documented on their respective websites."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "3136e6dd",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Here is a system prompt we have come up with which works well for granite-3.2-8b\n",
+    "granite_model=\"ibm-granite/granite-3.2-8b-instruct\"\n",
+    "granite_prompt=\"\"\"You are a helpful AI assistant with access to the tools listed next. When a tool is required to answer the user's query, respond with `<tool_call>` followed by a JSON object of the tool used. For example: `<tool_call> {\"name\":\"function_name\",\"arguments\":{\"arg1\":\"value\"}} </tool_call>`:The user will respond with the output of the tool execution response so you can continue with the rest of the initial user prompt (continue).\n",
+    "If a tool does not exist in the provided list of tools, notify the user that you do not have the ability to fulfill the request. \"\"\"\n",
+    "\n",
+    "# Here is a system prompt we have come up with which works well for llama-3.2-3b\n",
+    "llama_model=\"meta-llama/Llama-3.2-3B-Instruct\"\n",
+    "llama_prompt= \"\"\"You are a helpful assistant. You have access to a number of tools.\n",
+    "Whenever a tool is called, be sure return the Response in a friendly and helpful tone. Whenever a pod has an error query the vector DB on the error from the pods and return a summary of steps to take\"\"\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6bc8d3f8",
+   "metadata": {},
+   "source": [
+    "# Report errors and possible solutions from Documents to team via Slack"
```

Comment:
Rephrase as `Resolve and report any errors that might be happening on my OpenShift Cluster`